src/components/expression/expression-renderer.js

```diff
@@ -220,27 +293,110 @@ genotet.ExpressionRenderer.prototype.layout = function() {
   this.profileHeight_ = this.data.options.showProfiles ?
     this.DEFAULT_PROFILE_HEIGHT : 0;
 
-  this.svgHeatmap_
-    .attr('transform', genotet.utils.getTransform([0, 0]));
-
-  this.heatmapWidth_ = this.canvasWidth_ - this.geneLabelWidth_;
+  this.heatmapWidth_ = this.canvasWidth_;
   if (this.data.options.showGeneLabels) {
-    this.heatmapWidth_ -= this.LABEL_MARGIN_;
+    this.heatmapWidth_ -= this.LABEL_MARGIN + this.geneLabelWidth_;
+  } else if (this.data.options.showProfiles) {
+    this.heatmapWidth_ -= this.DEFAULT_PROFILE_MARGIN;
   }
-  this.heatmapHeight_ = this.canvasHeight_ - this.profileHeight_ -
-    this.conditionLabelHeight_;
+  this.heatmapHeight_ = this.canvasHeight_ -
+    this.profileHeight_ - this.DEFAULT_HEATMAP_GRADIENT_HEIGHT;
   if (this.data.options.showConditionLabels) {
-    this.heatmapHeight_ -= this.LABEL_MARGIN_;
+    this.heatmapHeight_ -= this.conditionLabelHeight_ +
+      this.LABEL_MARGIN + this.LABEL_DIFFERENCE;
   }
 
+  var heatmapData = this.data.matrix;
```

Comment:
The layout() function is a bit too bulky now. Consider modularizing it and create sub-functions for sub-routines (maybe later).